src/libraries/System.Diagnostics.Process/tests/ProcessWaitingTests.cs

```diff
@@ -143,6 +290,38 @@ public void SingleProcess_CopiesShareExitInformation()
             }
         }
 
+        [Fact]
+        public async Task SingleProcess_CopiesShareExitAsyncInformation()
+        {
+            Process p = CreateProcessLong();
+            p.Start();
+
+            Process[] copies = Enumerable.Range(0, 3).Select(_ => Process.GetProcessById(p.Id)).ToArray();
+
+            using (var cts = new CancellationTokenSource(0))
+            {
+                var token = cts.Token;
+                var ex = await Assert.ThrowsAnyAsync<OperationCanceledException>(() => p.WaitForExitAsync(token));
+                Assert.Equal(token, ex.CancellationToken);
+                Assert.False(p.HasExited);
+            }
+            p.Kill();
+            using (var cts = new CancellationTokenSource(WaitInMS))
+            {
+                await p.WaitForExitAsync(cts.Token);
+                Assert.True(p.HasExited);
+            }
+
+            using (var cts = new CancellationTokenSource(0))
+            {
+                foreach (Process copy in copies)
+                {
+                    await copy.WaitForExitAsync(cts.Token);
+                    Assert.True(copy.HasExited);
```

Comment:
This is a little confusing, may want to comment here as one might assume calling `WaitForExitAsync` with a cancelled token would throw.

Reply:
Added!